Plugins/NativeEngine/Source/ShaderCompilerOpenGL.cpp

```diff
@@ -36,12 +36,14 @@ namespace Babylon
 
             auto compiler = std::make_unique<spirv_cross::CompilerGLSL>(parser->get_parsed_ir());
 
+#if !ANDROID
             compiler->build_combined_image_samplers();
+#endif
 
             spirv_cross::CompilerGLSL::Options options = compiler->get_common_options();
 
-#ifdef ANDROID
-            options.version = 310;
+#if ANDROID
+            options.version = 300;
```

Comment:
Does this work? I seem to remember there was some issue with 300, like SPIRV-Cross had some sort of problem with that version.

Reply:
What do you mean by "in isolation"? It certainly works for ANDROID builds AFAICT.

Reply:
BTW, this change is what removes the binding attributes. Bindings are not supported until 310, but using 310 causes the bindings to output with locations of 0 for each one. WebGL2 conforms closely to OpenGL ES 3.0. We should use 3.0 unless there is a specific reason not to.

Reply:
By "in isolation," I meant were there other changes (for example, a different SPIRV-Cross submodule version) that weren't reflected here that caused this to work (realizing it's a draft and might not be complete). I don't remember exactly what it was, but _something_ led us to decide to use GLSL 310 instead of 300. If that was the bindings thing and it no longer applies, that's awesome!

Reply:
I don't remember/know for sure. But this change is tied with the removal of the the split and recombine in this file.